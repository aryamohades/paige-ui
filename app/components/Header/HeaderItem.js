import React from 'react';
import PropTypes from 'prop-types';

import { StyledLink, StyledButton, StyledButtonPrimary } from './itemStyles';

const HeaderItem = ({ primary, text, to, type }) => {
  let Wrapper = primary ? StyledButtonPrimary : StyledButton;

  if (type === 'link') {
    Wrapper = StyledLink;
  }

  return (
    <Wrapper to={to}>
      <span>{text}</span>
    </Wrapper>
  );
};

HeaderItem.propTypes = {
  primary: PropTypes.bool,
  text: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['link', 'button']),
};

HeaderItem.defaultProps = {
  type: 'link',
};

export default HeaderItem;
