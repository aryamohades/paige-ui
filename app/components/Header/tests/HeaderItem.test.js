import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import HeaderItem from '../HeaderItem';

describe('<HeaderItem />', () => {
  it('should match the snapshot', () => {
    const props = { text: 'test', to: '/' };
    const renderedComponent = renderer
      .create(
        <Router>
          <HeaderItem {...props} />
        </Router>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the correct text', () => {
    const props = { text: 'test', to: '/' };
    const renderedComponent = mount(
      <Router>
        <HeaderItem {...props} />
      </Router>,
    );
    expect(renderedComponent.find(HeaderItem).text()).toEqual(props.text);
  });
});
