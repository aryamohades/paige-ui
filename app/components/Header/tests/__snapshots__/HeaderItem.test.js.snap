// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderItem /> should match the snapshot 1`] = `
<a
  className="itemStyles__StyledLink-fAIfBS gvqALX"
  href="/"
  onClick={[Function]}
>
  <span>
    test
  </span>
</a>
`;
